### PR TITLE
[Doc] Add ray summit banner 2024

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

This PR adds a Ray Summit banner for 2024.

## Related issue number

No issue number.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/_static/css/custom.css [hunk 1]
```diff
@@ -65,8 +69,8 @@ h6 {
   background-color: transparent;
 }
 .bd-content:after {
-/*   Commenting the code below to make more modifications after 2.10 release */
-/*   background: linear-gradient(
+  /*   Commenting the code below to make more modifications after 2.10 release */
+  /*   background: linear-gradient(
```
Comment (aslonnie) [inline doc/source/_static/css/custom.css:73]: not sure where this comes from.. but have passed 2.10 now.

File: doc/source/_static/css/custom.css [hunk 2]
```diff
@@ -3,6 +3,7 @@
 * for more information. `important!` is needed below to override
 * dark/light theme specific values, which normally take precedence over the PST defaults.
 * */
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;900&family=Poppins&family=Roboto:wght@400;500;700&display=swap');
```
Comment (aslonnie) [inline doc/source/_static/css/custom.css:6]: why is this required now? and this not only import `Poppins` but also `Roboto`, which is not used in this css file? is it that important to use the `Poppins` font? It will make page loading slightly slower, right?

Comment (peytondmurray) [reply doc/source/_static/css/custom.css:6]: > is it that important to use the Poppins font? It will make page loading slightly slower, right?

So I'm in agreement about this - IMO we should not be loading a new font here for a single element either, nor do I think we should be introducing new colors rather than using existing Ray branding. But these designs were been passed to me from marketing, so I didn't think I had wiggle room to change this.

---

EDIT: After talking with Angelina about this and some other design considerations, it seems like Poppins is not needed.

File: doc/source/_static/css/custom.css (hunk 1 above)
Comment (peytondmurray) [reply doc/source/_static/css/custom.css]: Looks like these changes came from elsewhere; after rebasing they're gone.

File: doc/source/conf.py [hunk 3]
```diff
@@ -265,7 +265,7 @@ def render_svg_logo(path):
 # documentation.
 html_theme_options = {
     "use_edit_page_button": True,
-    "announcement": None,
+    "announcement": "Ray Summit unites the community. Submit a <b><a target='_blank' href='https://raysummit.anyscale.com/flow/anyscale/raysummit2024/landing/page/eventsite??utm_source=cfpraydocs_gl=1*949o36*_ga*Mzk2ODgxMTUwLjE3MTM4OTE0NzM.*_ga_T6EXHYG44V*MTcxNDA3ODUzNC43LjAuMTcxNDA3ODUzNC42MC4wLjEyMDk3NzQ2MjE'>session proposal</a></b> to share your experience.",
```
Comment (aslonnie) [inline doc/source/conf.py:268]:
```suggestion
    "announcement": "Ray Summit unites the community. Submit a <b><a target='_blank' href='https://raysummit.anyscale.com/flow/anyscale/raysummit2024/landing/page/eventsite?utm_source=cfpraydocs_gl=1*949o36*_ga*Mzk2ODgxMTUwLjE3MTM4OTE0NzM.*_ga_T6EXHYG44V*MTcxNDA3ODUzNC43LjAuMTcxNDA3ODUzNC42MC4wLjEyMDk3NzQ2MjE'>session proposal</a></b> to share your experience.",
```

maybe? @angelinalg ? 

@peytondmurray where do you get this URL from?

Comment (peytondmurray) [reply doc/source/conf.py:268]: @angelinalg passed this to me.

**Review by angelinalg — APPROVED** [review]

**Review by aslonnie — APPROVED** [review]
